src/cmd/plugin.rs

```diff
@@ -0,0 +1,138 @@
+/*  artifact: the requirements tracking tool made for developers
+ * Copyright (C) 2017  Garrett Berg <@vitiral, vitiral@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Lesser GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the Lesser GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * */
+
+use clap::{App, Shell};
+use std::fs;
+use dev_prefix::*;
+use types::*;
+use cmd::types::*;
+use cmd::matches::art_app;
+
+pub fn get_subcommand<'a, 'b>() -> App<'a, 'b> {
+    SubCommand::with_name("plugin")
+        .settings(&SUBCMD_SETTINGS)
+        .about(
+            "Command for integrating with external plugins, currently only \
+             supports generating shell completions",
+        )
+        .arg(
+            Arg::with_name("name")
+                .required(true)
+                .possible_values(&[
+                    "bash-completions",
```

Comment:
TODO: make these constants. I can tackle if you are busy.

```
static BASH_COMPLETIONS: &str = "bash-completions";
... etc
```

Reply:
I think we want `const` not `static`, with `const` the match expression can be
```rust
    BASH_COMPLETIONS => Plugin::Comp(Shell::Bash),
    FISH_COMPLETIONS => Plugin::Comp(Shell::Fish),
    ZSH_COMPLETIONS => Plugin::Comp(Shell::Zsh),
    PS_COMPLETIONS => Plugin::Comp(Shell::PowerShell),
    ...
```
as the `const` effectively inlines the constant. Otherwise we would need to do something like 
```rust
    x if x == BASH_COMPLETIONS => Plugin::Comp(Shell::Bash),
    x if x == FISH_COMPLETIONS => Plugin::Comp(Shell::Fish),
    x if x == ZSH_COMPLETIONS => Plugin::Comp(Shell::Zsh),
    x if x == PS_COMPLETIONS => Plugin::Comp(Shell::PowerShell),
    ...
```
I'm not certain what this becomes, but it seems less elegant for serving the same purpose.